crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,419 @@
+        .into()
+    }
+}
+
+impl From<transformers::Foreign<(BluesnapTxnType, Option<BluesnapSyncProcessingStatus>)>> for transformers::Foreign<enums::AttemptStatus> {
+    fn from(
+        item: transformers::Foreign<(BluesnapTxnType, Option<BluesnapSyncProcessingStatus>)>,
+    )-> Self {
+        let item = item.0;
+        let (item_txn_status, item_processing_status) = item;
+        match item_processing_status {
+            Some(status) => {
+                match status{
+                    BluesnapSyncProcessingStatus::Success => match item_txn_status {
+                        BluesnapTxnType::AuthOnly => enums::AttemptStatus::Authorized,
+                        BluesnapTxnType::AuthReversal => enums::AttemptStatus::Voided,
+                        BluesnapTxnType::AuthCapture
+                        | BluesnapTxnType::Capture
+                        | BluesnapTxnType::Refund => enums::AttemptStatus::Charged,
+                    },
+                    BluesnapSyncProcessingStatus::Pending
+                    | BluesnapSyncProcessingStatus::PendingMerchantReview => enums::AttemptStatus::Pending,
+                    BluesnapSyncProcessingStatus::Fail => enums::AttemptStatus::Failure,
+                    _ => enums::AttemptStatus::Charged,
+                }
+            },
+            _ => enums::AttemptStatus::Pending,
+        }
+        .into()
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapChargePaymentsResponse {
+    processing_info: BluesnapChargePaymentsProcessingInfoResponse,
+    transaction_id: String,
+    card_transaction_type: BluesnapTxnType,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapSyncPaymentsResponse {
+    processing_info: BluesnapSyncPaymentsProcessingInfoResponse,
+    transaction_id: String,
+    card_transaction_type: BluesnapTxnType,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Refund {
+    refund_transaction_id: String,
+    amount: String,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapChargePaymentsProcessingInfoResponse {
+    processing_status: Option<BluesnapChargeProcessingStatus>,
+    cvv_response_code: String,
+    authorization_code: String,
+    avs_response_code_zip: String,
+    avs_response_code_address: String,
+    avs_response_code_name:String,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapSyncPaymentsProcessingInfoResponse {
+    processing_status: Option<BluesnapSyncProcessingStatus>,
+    cvv_response_code: String,
+    authorization_code: String,
+    avs_response_code_zip: String,
+    avs_response_code_address: String,
+    avs_response_code_name:String,
+}
```

Comment:
These are same struct repeated, is it necessary or its redundant

Reply:
Actually processing_status has different case in both the responses, that's why had to declare redundant structs, with only processing_status' type varying

Reply:
Please suggest any refactoring if possible here

Reply:
I will pull the branch and try to find optimisation, you need not worry about this.
But if you are keen to do it, we can spend time on it.